components/table/test/table.test.js

```diff
@@ -102,6 +145,28 @@ describe('d2l-table-wrapper', () => {
 
 	});
 
+	describe('async-item-load', () => {
+		let elem, wrapper;
+		beforeEach(async() => {
+			elem = await fixture(`<${tag}></${tag}>`, {
+				awaitLoadingComplete:false
+			});
+			wrapper = elem.shadowRoot.querySelector('d2l-table-wrapper');
+			await oneEvent(wrapper, 'd2l-table-wrapper-layout-change');
+		});
+
+		it('change-event-fired', async() => {
+			await expect(wrapper._itemShowingCount).to.equal(2);
+		});
+
+		it('loads all items after single update', async() => {
+			setTimeout(elem._loadItemsAsync());
+			await oneEvent(wrapper, 'd2l-table-wrapper-layout-change');
+			await wrapper.updateComplete;
```

Comment:
Is this extra `await wrapper.updateComplete` necessary? If so, I wonder if we could delay firing the event until after the next render so that consumers/tests only need to await one thing.